scripts/fuzz.sh

```diff
@@ -0,0 +1,37 @@
+#!/bin/bash
+set -e
+
+# Simple wrapper around a libfuzzer test run, as much for
+# documentation as direct use.  The idea here is really simple: build
+# for the Zephyr "native_posix" board (which is a just a x86
+# executable for the build host, not an emulated device) and run the
+# resulting zephyr.exe file.  This specifies a "fuzz_corpus" directory
+# to save the seeds that produce useful coverage output for use in
+# repeated runs (these are not particularly large, we might consider
+# curating and commiting such a seed directory to the tree).
+#
+# The tool will run until it finds a failure condition.  You will see
+# MANY errors on stdout from all the randomized input.  Don't try to
+# capture this, either let it output to a terminal or arrange to keep
+# only the last XXX lines after the tool exits.
+#
+# The only prerequisite to install is a clang compiler on the host.
+# Versions 12+ have all been observed to work.
+#
+# You will need the kconfigs specified below for correct operation,
+# but can add more at the end of this script's command line to
+# duplicate configurations as needed.
+
+export ZEPHYR_BASE=$SOF_TOP/../zephyr
```

Comment:
I haven't used ZEPHYR_BASE for a long time, are you sure it's required in this case? Can this script be run outside a west workspace?

Reply:
It's not needed in the default setup.  But I have like a thousand zephyr trees floating around and do dumb stuff like symlinking module directories across them.  If you have ZEPHYR_BASE set to the **wrong** tree, you get very funny results (like, the build works, but uses the wrong SOF tree!).  So I guess I'd see this as a best practices kind of thing to ensure that the tree this script is running from is the one being built.